imap_processing/codice/codice_l1a.py

```diff
@@ -94,13 +98,14 @@ def create_science_dataset(self, packets: list) -> xr.Dataset:
         -------
         xr.Dataset
             ``xarray`` dataset containing the science data and supporting metadata
-
-        # TODO: Pull out common code and put in codice.utils alongside
-        # create_hskp_dataset()
-        # TODO: Resolve "Python into too large to convert to C long" error
         """
         epoch = xr.DataArray(
-            [calc_start_time(packets[0].data["SHCOARSE"].raw_value)],
+            [
+                calc_start_time(
+                    packets[0].data["ACQ_START_SECONDS"].raw_value,
+                    launch_time=np.datetime64("2010-01-01T00:01:06.184", "ns"),
+                )
+            ],
```

Comment:
Do you want a start time per packet? Otherwise, I think this is just a single epoch value as your coordinate.
```suggestion
            [
                calc_start_time(
                    packet.data["ACQ_START_SECONDS"].raw_value,
                    launch_time=np.datetime64("2010-01-01T00:01:06.184", "ns"),
                ) for packet in packets
            ],
```

Reply:
Thanks, nice catch. I think this is a relic from some debugging-past.